### PR TITLE
Add documentation about Simulate() in Run() function

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR is a followup to #2526 and adds a bit more context explaining how run is triggered during simulate. 

@JoseJVS let me know if this should be expanded upon, and if you think the `Simulate` docstring should also include more details.

### CODE REVIEW

**Review by JoseJVS — APPROVED** [review]
Everything looks good!

File: pynest/nest/lib/hl_api_simulation.py [hunk 1]
```diff
@@ -105,11 +109,15 @@ def Run(t):
...
-    """Calibrate the system before a `Run` call. Not needed for `Simulate`.
+    """Calibrate the system before a `Run` call.
+    `Prepare` is automatically triggered at beginning of call to `Simulate`:
+
+    Simulate(t): t' = t/m; Prepare(); for _ in range(m): Run(t'); Cleanup()
 
     Call before the first `Run` call, or before calling `Run` after changing
     the system, calling `SetStatus` or `Cleanup`.
 
+
```
Comment (BenjaminPoulain) [inline pynest/nest/lib/hl_api_simulation.py:120]: Here and below: is this intentional?

There is only a single blank line above titles elsewhere.
```suggestion
```

Comment (jessica-mitchell) [reply pynest/nest/lib/hl_api_simulation.py]: No this is unintentional, I fixed this and also I noticed there should be a new line after the title of the docstring.

File: pynest/nest/lib/hl_api_simulation.py [hunk 2]
```diff
@@ -49,6 +49,10 @@
 def Simulate(t):
     """Simulate the network for `t` milliseconds.
 
+    `Simulate` triggers `Prepare`, `Run`, and `Cleanup` following this formula:
+
+    Simulate(t): t' = t/m; Prepare(); for _ in range(m): Run(t'); Cleanup()
```
Comment (heplesser) [inline pynest/nest/lib/hl_api_simulation.py:54]:
```suggestion
    `Simulate(t)` runs `Prepare()`, `Run(t)`, and `Cleanup()` in this order.
```

Comment (heplesser) [reply pynest/nest/lib/hl_api_simulation.py:54]: Also add `Prepare, Run, Cleanup` to the "See also" section below; I cannot add it as a suggestion, unfortunately.

File: pynest/nest/lib/hl_api_simulation.py [hunk 3]
```diff
@@ -78,6 +82,7 @@ def Run(t):
 
     Call between `Prepare` and `Cleanup` calls, or within a
     ``with RunManager`` clause.
+    `Run` is automatically triggered during call to `Simulate`:
 
     Simulate(t): t' = t/m; Prepare(); for _ in range(m): Run(t'); Cleanup()
```
Comment (heplesser) [inline pynest/nest/lib/hl_api_simulation.py:87]:
```suggestion
    Call between `Prepare` and `Cleanup` calls, or within a
    ``with RunManager`` clause.  `Run(t)` is called once by each call to `Simulate(t)`.
```

File: pynest/nest/lib/hl_api_simulation.py [hunk 4]
```diff
@@ -105,7 +110,11 @@ def Run(t):
 
 @check_stack
 def Prepare():
-    """Calibrate the system before a `Run` call. Not needed for `Simulate`.
+    """Calibrate the system before a `Run` call.
+
+    `Prepare` is automatically triggered at beginning of call to `Simulate`:
+
+    Simulate(t): t' = t/m; Prepare(); for _ in range(m): Run(t'); Cleanup()
```
Comment (heplesser) [inline pynest/nest/lib/hl_api_simulation.py:117]:
```suggestion
    """Calibrate the system before a `Run` call.

    `Prepare` is automatically called by `Simulate` and `RunManager`.
```

Also add `Simulate, RunManager` to See Also below.

File: pynest/nest/lib/hl_api_simulation.py [hunk 5]
```diff
@@ -121,7 +130,11 @@ def Prepare():
 
 @check_stack
 def Cleanup():
-    """Cleans up resources after a `Run` call. Not needed for `Simulate`.
+    """Cleans up resources after a `Run` call.
+
+    `Cleanup` is automatically triggered at the end of call to `Simulate`:
+
+    Simulate(t): t' = t/m; Prepare(); for _ in range(m): Run(t'); Cleanup()
```
Comment (heplesser) [inline pynest/nest/lib/hl_api_simulation.py:137]:
```suggestion
    """Cleans up resources after a `Run` calls.

    `Cleanup` is automatically called by `Simulate` and `RunManager` .
```

Also add `Simulate, RunManager` to See Also below.

**Review by heplesser — CHANGES_REQUESTED** [review]
@jessica-mitchell  Thanks! I think the comments should be simplified, see my comments below.

**Review by heplesser — APPROVED** [review]

**Pull request merged**